### PR TITLE
Prepare for release 1.0.7

### PR DESCRIPTION
**Pull request by Mandur**
# PR for issue #359 

## What is being addressed

To prepare release 1.0.7, we need to ensure this get deployed by default by the arm template

## How is this addressed

* Update the ARM teamplate
* Update migration guide

### CODE REVIEW

File: Docs/upgrade.md [hunk 1]
```diff
@@ -1,6 +1,25 @@
 # Upgrade LoRaWAN to a new version
 <!-- allow github to generate 'copy' buttons for coded URLs -->
 <!-- markdownlint-disable MD040 -->
+## Release 1.0.7
+
+> Note that this version switch IoT Edge to use version 1.2 of IoT Edge that include major changes on the edge. Please follow this [documentation](https://docs.microsoft.com/azure/iot-edge/how-to-update-iot-edge?view=iotedge-2020-11&tabs=linux) to upgrade IoT Edge to 1.2.
```
Comment (p-schuler) [inline Docs/upgrade.md:6]: nit: not sure how you do this usually, but the official name would be Azure IoT Edge.

Comment (p-schuler) [inline Docs/upgrade.md:6]:
```suggestion
> This version starts using Azure IoT Edge 1.2 which includes major changes. Please follow [Update IoT Edge](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-update-iot-edge?view=iotedge-2020-11&tabs=linux) to upgrade to Azure IoT Edge 1.2.
```

File: Docs/upgrade.md [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+
+To update from version 1.0.6, 1.0.5, 1.0.4 or 1.0.3 you can follow the below instructions. If you want to update manually from a version prior to 1.0.3, please refer to the instructions in the [Release 1.0.3](#Release-1.0.3) section below.
+
+### Updating from release after 1.0.3
```
Comment (p-schuler) [inline Docs/upgrade.md:10]: nit
```suggestion
### Updating from release post 1.0.3
```

**Comment by p-schuler** [comment]
From reading the instructions, it's not clear to me, what I need to do. Do I have to follow https://docs.microsoft.com/azure/iot-edge/how-to-update-iot-edge?view=iotedge-2020-11&tabs=linux or can I simply update the version on my modules in IoT hub? Sorry, that wasn't clear to me.

**Review by p-schuler — CHANGES_REQUESTED** [review]
The instructions are not obvious to me in terms of what steps are required to take to update.

File: Docs/upgrade.md (hunk 1 above)
Comment (Mandur) [reply Docs/upgrade.md:6]: Right, much better. I will just keep the URL langage agnostic

Comment (p-schuler) [reply Docs/upgrade.md]: By URL language you mean the title of the document? It's considered best practice to use the title in the links.

File: Docs/upgrade.md [hunk 3]
```diff
@@ -1,12 +1,35 @@
...
-## Release 1.0.6
+## Release 1.0.7
+
+To update from version 1.0.6, 1.0.5, 1.0.4 or 1.0.3 you can follow the below instructions. If you want to update manually from a version prior to 1.0.3, please refer to the instructions in the [Release 1.0.3](#Release-1.0.3) section below.
+
+### Update the IoT Edge security daemon when upgrading from IoT Edge 1.1 (release prior to 1.0.6)
+
+Since release 1.0.6, the starter kit uses Azure IoT Edge version 1.2 which includes major changes the the IoT Edge Security daemon. Please follow this documentation to [Update IoT Edge](https://docs.microsoft.com/azure/iot-edge/how-to-update-iot-edge?view=iotedge-2020-11&tabs=linux) to upgrade Azure IoT Edge to 1.2.
+
+### Updating from release post 1.0.3
```
Comment (p-schuler) [inline Docs/upgrade.md:12]:
```suggestion
### Updating IoT Edge LoRaWAN starterkit from release post 1.0.3
```

**Review by p-schuler — APPROVED** [review]
LGTM

**Pull request merged**